common/djangoapps/student/views.py

```diff
@@ -707,9 +707,31 @@ def dashboard(request):
         course_runs_for_course = get_course_runs_for_course(str(course_entitlement.course_uuid))
         enrollable_course_runs = []
 
+        # Only show published course runs that can still be enrolled and upgraded
+        now = datetime.datetime.now(UTC)
         for course_run in course_runs_for_course:
+            # Only courses that have not ended will be displayed
+            run_start = course_run.get('start')
+            run_end = course_run.get('end')
+            is_running = start and (not run_end or (dateutil.parser.parse(run_end) > now))
+
+            # Only courses that can currently be enrolled in will be displayed
+            enrollment_start = course_run.get('enrollment_start')
             enrollment_end = course_run.get('enrollment_end')
-            if not enrollment_end or (dateutil.parser.parse(enrollment_end) > datetime.datetime.now(UTC)):
+            can_enroll = (not enrollment_start or dateutil.parser.parse(enrollment_start) < now
```

Comment:
Is there a method in the CourseEnrollemnt Model that you can use for this already in existence? I am looking now in the code.